🛠 Tooling: Typed typescript-eslint rules are running on files without type information

### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/template-typescript-node-package/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

Discovered in https://github.com/JoshuaKGoldberg/emojisplosion/pull/53 -> https://github.com/JoshuaKGoldberg/emojisplosion/actions/runs/3633095030/jobs/6129716225:

```plaintext
/home/runner/work/emojisplosion/emojisplosion/script/setup.js
...

/home/runner/work/emojisplosion/emojisplosion/vitest.config.ts
  0:0  error  Parsing error: ESLint was configured to run on `<tsconfigRootDir>/vitest.config.ts` using `parserOptions.project`: <tsconfigRootDir>/tsconfig.json

/home/runner/work/emojisplosion/emojisplosion/webpack.config.js
...
```

The issue is `tsconfig.json` excludes those files. I'll set up a `tsconfig.eslint.json` per https://typescript-eslint.io/linting/troubleshooting#i-get-errors-telling-me-eslint-was-configured-to-run--however-that-tsconfig-does-not--none-of-those-tsconfigs-include-this-file.

### Additional Info

_No response_

--- a/.eslintrc.cjs
+++ b/.eslintrc.cjs
@@ -23,6 +23,13 @@ module.exports = {
 				"plugin:@typescript-eslint/strict",
 			],
 			files: ["**/*.{ts,tsx}"],
+			parserOptions: {
+				project: "./tsconfig.eslint.json",
+			},
+			rules: {
+				// These off-by-default rules work well for this repo and we like them on.
+				"deprecation/deprecation": "error",
+			},
 		},
 		{
 			files: "*.json",
@@ -41,10 +48,6 @@ module.exports = {
 		},
 	],
 	parser: "@typescript-eslint/parser",
-	parserOptions: {
-		tsconfigRootDir: __dirname,
-		project: ["./tsconfig.json"],
-	},
 	plugins: [
 		"@typescript-eslint",
 		"deprecation",
@@ -57,7 +60,6 @@ module.exports = {
 	root: true,
 	rules: {
 		// These off-by-default rules work well for this repo and we like them on.
-		"deprecation/deprecation": "error",
 		"no-only-tests/no-only-tests": "error",
 		"simple-import-sort/exports": "error",
 		"simple-import-sort/imports": "error",

--- a/tsconfig.eslint.json
+++ b/tsconfig.eslint.json
@@ -1,0 +1,4 @@
+{
+	"extends": "./tsconfig.json",
+	"include": ["."]
+}

